helix-term/src/ui/picker.rs

```diff
@@ -256,9 +256,38 @@ impl<T: 'static> Component for FilePicker<T> {
         }
     }
 
-    fn handle_event(&mut self, event: Event, ctx: &mut Context) -> EventResult {
+    fn handle_event(&mut self, event: Event, cx: &mut Context) -> EventResult {
+        let key_event = match event {
+            Event::Key(event) => event,
+            _ => return self.picker.handle_event(event, cx),
+        };
+
+        let mut filter = |d| {
+            self.picker.options.retain(|item| {
+                (self.file_fn)(cx.editor, item)
+                    .map(|fl| fl.0.starts_with(d))
+                    .unwrap_or(false)
+            });
+            self.picker.filters.clear();
+            self.picker.score();
+        };
+
+        match key_event.into() {
+            ctrl!('a') => {
+                if let Ok(cwd) = std::env::current_dir() {
+                    filter(cwd.as_path());
+                }
+                EventResult::Consumed(None)
+            }
+            ctrl!('f') => {
+                if let Some(cwd) = doc!(cx.editor).path().and_then(|p| p.parent()) {
+                    filter(cwd);
+                }
+                EventResult::Consumed(None)
+            }
```

Comment:
Hmm, this seemed oddly specific. Do you know any prior art that do something similar? I don't see something like this in doom emacs IIRC or maybe I didn't dig deep, also stuff like these are hard to discover. And what is your usecase?

I wonder if it would be good to `C-r %` to paste the current file and `C-w` to remove the last file path to do the same without doing something so specific?